src/index.js

```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+module.exports = {
+  Stop: require('./circuit/hop'),
+  Dialer: require('./dialer'),
+  multicodec: require('./multicodec'),
+  tag: 'Circuit'
+}
```

Comment:
Yep, this is handled by the swarm using the standard transport init flow - https://github.com/libp2p/js-libp2p-swarm/blob/master/src/transport.js#L82.